### DISCUSSION TITLE
How to open files/Floder on mobile hard discs? | Mac OS code-server

### DISCUSSION THREAD

**Discussion by lindw-hub** [coder/code-server#4206] [Q&A] [answered]
I want to code with my iPad. So I install code-server on my Mac mini through `brew`.
It is ok to open any files/Folders on my Mac mini Macintosh HD through Web apps on my iPad.
But, I can not open any files/folders on my mobile hard disc which is connected to my Mac mini, with the warning 'Operation not permitted'. However, I can attach the files/folders on my mobile hard disc through ssh using Termius from my iPad.
I also tried to add the 'code-server' to Full Disk Access, but it did not work.

What should I do next? Can anyone give me some advice?  Thanks!!

**Comment by jsjoeio** [comment]
Talking this out loud so I understand your setup.

- code-server is installed on your Mac mini with `brew install code-server`
- you run it locally on the Mac mini with `code-server` (maybe with `--link`?)
- you access it on your iPad somehow

> But, I can not open any files/folders on my mobile hard disc which is connected to my Mac mini

This part I'm confused about. Are you saying you can't access files/folders on the iPad's hard disc?

**Reply by lindw-hub** [reply]
My setup:
- installed on my Mac mini with `brew install code-server`
- code-server was started with `brew services start code-server`
- Because accessing it with iPad did not work as my expectation, I also tried to access it on Google Chrome with `http://localhost:8080` on the Mac mini.
But the problem remains. I can not open folders on any external hard discs that are connected to my Mac mini.

- I tried to open the Datadisk3 which is an external hard disc connected to the Mac mini.
![figure_1632228420](https://user-images.githubusercontent.com/67275356/134196389-f1b1aa46-a7c2-4c59-b25d-ae4116823362.png)

- However, It did not reveal any files and folders in the external hard disc. While I want to list the files under the hard disc, it warns "operation not permitted. 
![figure_1632228600](https://user-images.githubusercontent.com/67275356/134196467-e3248722-9c0a-45f1-8d21-c48502a4d8ef.png)

It seems that I am not allowed to access the external hard disc by visiting `http://127.0.0.1:8080`. 
I can access the external hard discs through SSH and open the files in hard discs with Google  Chrome. I also tried to start she code-server with `sudo brew services start code-server`, and it still did not work. 

I'd like to know if it's a problem with my computer permissions or code-server settings. Thanks!!! 🙏🙏

**Reply by jsjoeio** [reply]
Super super helpful! Thanks for the details + screenshots.

> I also tried to access it on Google Chrome with http://localhost:8080 on the Mac mini.
But the problem remains. I can not open folders on any external hard discs that are connected to my Mac mini.

This leads me to believe that this is a computer issue related to the external hard discs. Are they password-protected? 

Can you open the folders on the Mac mini directly? (not inside code-server but with Finder)

What happens if you run `code-server` not through `brew services`? Is it the same result?

Any ideas @code-asher?

**Reply by lindw-hub** [reply]
Yes, It works!  I stopped code-server through `brew services stop code-server`. Then I restart the code-server through `code-server` as you suggested. It works! The folders on the external hard discs can be opened in the code-server. 

This reminds me that the command `brew` cannot be executed by `sudo` or root, which may cause this problem. 

Thanks again for your kind help!!! 🙏🙏🙏

**Reply by jsjoeio** [reply]
Wow! That was a total guess (I  don't use `brew services` so I didn't know that). 

> This reminds me that the command brew cannot be executed by sudo or root, which may cause this problem.

TIL!

Hooray! Go team!

**Comment by jsjoeio** [comment, marked as answer]
Don't run code-server with `brew services`!

> Yes, It works! I stopped code-server through brew services stop code-server. Then I restart the code-server through code-server as you suggested. It works! The folders on the external hard discs can be opened in the code-server.

> This reminds me that the command brew cannot be executed by sudo or root, which may cause this problem.
